semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -733,16 +760,21 @@ let rec stmt_aux env st =
       ss @ [ mk_s (If (tok, e', st1, st2)) ]
   | G.Switch (_, _, _) -> todo (G.S st)
   | G.While (tok, e, st) ->
+      let cont_label_s, break_label_s, st_env = mk_st_env_and_labels env tok in
```

Comment:
The original 'env' is used after, and because the translation has side effects (modifying the ref in the env), there are cases where I can't just switch around the order of calls.